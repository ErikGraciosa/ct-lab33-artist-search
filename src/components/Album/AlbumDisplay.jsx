import React from 'react';
import AlbumItem from './AlbumItem';
import PropTypes from 'prop-types';

function AlbumDisplay({ albums }) {
  return (
    <div>
      {albums.map(album => <AlbumItem key={album.id} {...album}/>)}
    </div>
  );
}

AlbumDisplay.propTypes = {
  albums: PropTypes.array
};

export default AlbumDisplay;
