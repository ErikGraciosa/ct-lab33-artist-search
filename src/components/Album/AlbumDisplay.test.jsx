import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AlbumDisplay from './AlbumDisplay';
import { MemoryRouter } from 'react-router-dom';

describe('AlbumDisplay component', () => {

  const albums = [{
    id: '1234'
  }];

  afterEach(() => cleanup());
  it.only('renders AlbumDisplay', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <AlbumDisplay albums={albums} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
