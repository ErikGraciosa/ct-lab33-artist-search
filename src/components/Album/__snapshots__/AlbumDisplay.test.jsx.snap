// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDisplay component renders AlbumDisplay 1`] = `
<DocumentFragment>
  <div>
    <div>
      <a
        href="/album/1234"
      >
        <figure>
          <img
            alt="picture"
            src="http://coverartarchive.org/release/1234/front"
          />
          <figcaption />
          <figcaption />
        </figure>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
